### PR TITLE
Publishig interactive data documents v1

### PR DESCRIPTION
**Pull request by daigotanaka**
@coffeeandchocolate: Please proof read and comment on this pull request. Thanks.

HTML version:
http://daigotanaka.org/publishing-interactive-data-documents-draft

### CODE REVIEW

File: publishing-interactive-data-documents.Rmd [hunk 1]
```diff
@@ -0,0 +1,234 @@
+---
+title: "Publishing interactive data documents"
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: test comment

Comment (daigotanaka) [inline publishing-interactive-data-documents.Rmd]: Yay, works :)

File: publishing-interactive-data-documents.Rmd [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+twitter_chart$series(name="Japanese", data = ja_values)
+twitter_chart$legend(symbolWidth = 80)
+twitter_chart$tooltip(pointFormat=get_point_format(y="Impressions", z="Engagements"))
+```
+
+## Publishing interactive data documents
+
+![Interactive chart](https://farm8.staticflickr.com/7341/15834649053_ca6a990af4_o.png)
+
+When presented effectively, an interactive chart helps audience to grasp the
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "an interactive chart helps the audience..."

File: publishing-interactive-data-documents.Rmd [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+overview and in-depth look of complex data. In this article, I will show
+an example of using interactive charts effectively. I also introduce R markdown
+as one of the most efficient ways of authoring data rich documents.
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "data-rich documents"

File: publishing-interactive-data-documents.Rmd [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+### Motivating example
+
+Visualizing data lets us grasp the birds-eye view of data. Suppose we made many
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "lets us get a bird's-eye view of data" or "lets us see a bird's-eye view of data"

You get or see a view, not grasp it. :)

File: publishing-interactive-data-documents.Rmd [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+observations of ozone and temperature to learn about air
+quality
+```{r}
+footnote("airquality")
+```
+. We can plot temperatures (F) and ozone level (ppb) on
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "We can plot temperature"

File: publishing-interactive-data-documents.Rmd [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+x and y axes, respectively as in Fig.  `r fn()`. Just by looking at the plot, we
+gain a little insight on the relationship between the two measurements.
+
+
+```{r, html.cap=fn("Ozone level and temperature")}
+# Plot air quality
+ozone_temp
+```
+
+More sophisticated visualization can present higher dimensional data on a flat
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: Not sure if you mean "A more sophisticated visualization" (meaning that in this specific case, something more sophisticated will help) or "More sophisticated visualizations" (meaning that in general, something more sophisticated will help).

"higher-dimensional data"...

File: publishing-interactive-data-documents.Rmd [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+paper: it explains not only the relationship between x and y, but also y and z,
+z and x, and so on. We can add the measurements of wind speed (mph) to our
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "We can add wind speed measurements (mph)..."

File: publishing-interactive-data-documents.Rmd [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+plot (Fig. `r fn()`).
+
+```{r, html.cap=fn("Ozone level, temperature, and wind speed. Lighter blue indicates higher wind speed.")}
+# Plot with wind speed
+ozone_temp_wind
+```
+
+In the plot, I showed the wind temperature in shades of blue. Lighter blue
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "A lighter blue..."

File: publishing-interactive-data-documents.Rmd [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+indicates higher wind speed. We can observe that the wind speed tends to be
+low when the ozone level is high.
+
+After grasping the overview of the data, we may want to look into the
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "After getting an overview of the data..."

It's not wrong to say "grasping", just weird.

File: publishing-interactive-data-documents.Rmd [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+details of particular parts of the data. What if we want to know the date of the
+observation of the highest point of ozone level? The graph on a paper is not
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "What if we want to know when the highest ozone level was observed?"

Too many prepositional phrases in original sentence.

File: publishing-interactive-data-documents.Rmd [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+ideal place to make the transition between the overview and in-depth look.
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: Not sure what you're saying in this sentence.

File: publishing-interactive-data-documents.Rmd [hunk 12]
```diff
@@ -0,0 +1,234 @@
...
+```
+
+### Interactive chart with R-markdown
+
+The interactive visualizations on New York Times articles
+```{r}
+footnote("nyt")
+```
+are the sophisticated
+example you may be already familiar.
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "You may already be familiar with the sophisticated interactive visualizations on New York Times articles."

File: publishing-interactive-data-documents.Rmd [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+
+It used to require a set of skills such as database (e.g. SQL),
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "It used to require skills such as database back-end processing (...) and front-end web application coding (...) to create such interactive visualizations."

File: publishing-interactive-data-documents.Rmd [hunk 14]
```diff
@@ -0,0 +1,234 @@
...
+back-end processing (Python, Ruby, nose.js, and etc), front-end web
+application coding (JavaScript, HTML, CSS) to create such interactive
+visualizations.
+
+Thanks to the recent developments such as
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "Thanks to recent developments..."

"data-rich documents..."

File: publishing-interactive-data-documents.Rmd [hunk 15]
```diff
@@ -0,0 +1,234 @@
...
+[R markdown](http://rmarkdown.rstudio.com/).
+
+With R markdown, one can:
+
+- Load and transform data from multiple sources
+- Run statistical analysis
+- Produce figures
+- Write narratives
+
+Within a single text file. Having to hop around different programs to do those
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "Having to switch between different programs to complete each of those tasks can be very disruptive to your thought process."

File: publishing-interactive-data-documents.Rmd [hunk 16]
```diff
@@ -0,0 +1,234 @@
...
+activities can be very disruptive to thought process. With R markdown, we
+can focus on the research without being distracted by how each tool works.
+
+### R-markdown example: Twitter impressions and engagements
+
+Figure `r fn()` is a demonstration of using R-markdown to visualize my twitter
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: Check capitalization: Twitter, not twitter

And be sure if it's "R Markdown" or "R-markdown" or "R-Markdown".

File: publishing-interactive-data-documents.Rmd [hunk 17]
```diff
@@ -0,0 +1,234 @@
...
+activities in 2014. X axis is time. Y axis is the number of impressions (i.e.
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "The x-axis represents time. The y-axis represents the number of impressions..."

File: publishing-interactive-data-documents.Rmd [hunk 18]
```diff
@@ -0,0 +1,234 @@
...
+how many people saw each of my tweets.) It is a bubble chart, and the size of
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: Delete "It is a bubble chart, and...." and just start with "The size of each bubble..."

File: publishing-interactive-data-documents.Rmd [hunk 19]
```diff
@@ -0,0 +1,234 @@
...
+each bubble represents the number of engagements (i.e. the total number of
+viewer's activities such as clicking the profile or links, and expanding the
+tweets.)
+
+```{r,  html.cap=fn("Interactive chart: Daigo's twitter impressions and engagements in 2014. The size of the bubble coresponds to the number of engagements. Hover the mouse cursor over to reveal the details.")}
+renderChart(twitter_chart, chartId="twitter", include_assets = TRUE, cdn = TRUE)
+```
+
+Bubble chart is one of the effective methods to visualize the relationship
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "A bubble chart like this is one way to effectively visualize the relationship between three variables."

File: publishing-interactive-data-documents.Rmd [hunk 20]
```diff
@@ -0,0 +1,234 @@
...
+between three variables. The chart above is also colored to distinguish
+the tweets in English and Japanese. With this presentation, you may easily
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "The chart above is also colored to distinguish between tweets in English and Japanese."

File: publishing-interactive-data-documents.Rmd [hunk 21]
```diff
@@ -0,0 +1,234 @@
...
+notice that the majority of my tweets are in English.
+
+After getting the overview of how the impressions, the engagements and frequency
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "After getting an overview of the impressions, engagements, and frequency of tweets in each language, I may want to find out which tweets actually got very high or low impressions or engagements."

File: publishing-interactive-data-documents.Rmd [hunk 22]
```diff
@@ -0,0 +1,234 @@
...
+of the tweets in each language, I may want to find out what tweet actually got
+very high or low impressions or
+engagements. In this chart, hovering the mouse over a bubble will reveal the
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: Delete "In this chart," and just say "Hovering your mouse over a bubble..."

File: publishing-interactive-data-documents.Rmd [hunk 23]
```diff
@@ -0,0 +1,234 @@
...
+content of each tweet. Each group can be toggled on and off, and the chart
+adjusts the zoom automatically.
+
+Once data is prepared, it only took 8 lines of R code to generate the chart
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "Once the data were prepared..."

Data is actually a plural noun.

File: publishing-interactive-data-documents.Rmd [hunk 24]
```diff
@@ -0,0 +1,234 @@
...
+with the help of rCharts package and
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "...with the help of the rCharts package..."

File: publishing-interactive-data-documents.Rmd [hunk 25]
```diff
@@ -0,0 +1,234 @@
...
+
+<script src="http://gist-it.appspot.com/https://github.com/daigotanaka/essays/blob/master/publishing-interactive-data-documents.Rmd?slice=82:90&footer=0">
+</script>
+
+The entire source of this article is
+[available for viewing](https://raw.githubusercontent.com/daigotanaka/essays/master/publishing-interactive-data-documents.Rmd).
+
+### Conclusions
+
+With the help of packages such as rCharts, it is now much easier to publish data
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "data-rich documents"

File: publishing-interactive-data-documents.Rmd [hunk 26]
```diff
@@ -0,0 +1,234 @@
...
+rich documents. R markdown let one focus on the data analysis and production of
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "R Markdown lets us focus on the data..."

You've been using "we" and "I" so it's a little weird to start using the impersonal "one" right now.

File: publishing-interactive-data-documents.Rmd [hunk 27]
```diff
@@ -0,0 +1,234 @@
...
+the manuscript without requiring in-depth knowledge of programming languages.
+Properly presented interactive charts help the online publishers to communicate
```
Comment (coffeeandchocolate) [inline publishing-interactive-data-documents.Rmd]: "...help online publishers communicate data and their interpretations effectively."

**Pull request merged**